src/todos/todos.service.spec.ts

```diff
@@ -0,0 +1,64 @@
+import { Test, TestingModule } from '@nestjs/testing'
+import { TodosService } from './todos.service'
+import { TodosDatabaseService } from './todos.database.service'
+import type { Todo } from './schemas/todos.schema'
+import { createMock } from '@golevelup/ts-jest'
+
+describe('TodosService', () => {
```

Comment:
Let's extend tests a bit, to ensure that not only correct methods have been called inside; but also that the correct value is returned.

Reply:
[Done](https://github.com/vladislav-shkutovich/NestJS-Todo-API/pull/4/commits/16d9fd02fe6039f0082bd470fe50af714ab0a0ec) using `jest.spyOn()`
Hopefully it's a one of the correct approaches

Reply:
Yeah, that's a good choice.
The only comment here — it's better to have one test func for one test case (ie separate test for external call; separate test for returned value; separate test for everything). + grouping tests by function name.
So, in general, it should look like this:
```TS
describe('TodosService', () => {
  describe('create()', () => {
    it('should call TodosDatabaseService.create() with correct arguments', async () => {});
    it('should return correct value', async () => {});
  });
  describe('getAll()', () => {});
  describe('delete()', () => {});
});
```

Reply:
Great idea, thanks!
Fixed [here](https://github.com/vladislav-shkutovich/NestJS-Todo-API/pull/4/commits/48c2cb8df850bde2e38021cd3fc1d73ebf2e0488)